R/labels.R

```diff
@@ -87,28 +102,141 @@ uses_labels <- function(p, ...) {
     )
   }
 
+  default_labels <- purrr::map_lgl(args, inherits, ".default_label")
+
+  args[default_labels] <- purrr::map(
+    names(args)[default_labels], ~ unlist(default_label(p, .))
+  )
+
   if (!all(is_scalar_string_or_null(args))) {
     stop(
       "All inputs to `...` must be character vectors of length 1 or `NULL`.",
       call. = FALSE
     )
   }
 
-  if (is.null(names(args)) || any(names(args) == "")) {
-    stop("All inputs to `...` must be named.", call. = FALSE)
+  if (is.null(names(args))) {
+    names(args) <- rep("",    length(args))
+    named       <- rep(FALSE, length(args))
+  } else {
+    named <- names(args) != ""
   }
 
-  labels <- get_labels(p, names(args))
+  result         <- logical(length(args))
+  result[!named] <- check_labels_set(p, args[!named])
+  result[named]  <- check_labels_match(p, args[named])
+  names(result)  <- coalesce_chr(names(args), args)
+  result
+}
+
+#' What is the default label for a plot aesthetic?
+#'
+#' @family functions for checking labels
+#' @export
+#'
+#' @examples
+#' require(ggplot2)
+#'
+#' p <- ggplot(data = mpg, mapping = aes(x = displ, y = hwy)) +
+#'   geom_point(mapping = aes(color = class, shape = drv)) +
+#'   geom_smooth() +
+#'   labs(title = "My plot", x = "Weight", y = "MPG", color = NULL)
+#'
+#' # Returns the label the ggplot would create by default for an aesthetic
+#' default_label(p, "x")
+#' default_label(p, c("x", "y"))
+#' default_label(p)
+#'
+#' # If an aesthetic does not exist, returns NULL
+#' default_label(p, "size")
+#'
+#' # If an aesthetic has no default, returns NULL
+#' default_label(p, "title")
+#'
+#' # The colo(u)r aesthetic can be matched with or without a u
+#' default_label(p, "color")
+#' default_label(p, "colour")
+#' @param p A [ggplot][ggplot2::ggplot] object
+#' @param aes If `aes` is a [character] vector, returns only the labels
+#'   corresponding to the included aesthetics.
+#'   Defaults to [`NULL`], which returns all labels.
+#'
+#' @return A named [list] in which each element is a [character] string
+#'   or [`NULL`]
```

Comment:
It'd be better to document why/when the return value is character or NULL and how to interpret those values. You could easily pull from the comments in the examples

Reply:
Resolved in e856502